app/submitter/submitter.py

```diff
@@ -39,8 +40,18 @@ def send_message(self, message: str, tx_id: str, case_id: str) -> bool:
 
         # DEFAULT_RETRY is not idempotent.
         # However, this behaviour was deemed acceptable for our use case.
-        blob.upload_from_string(str(message).encode("utf8"), retry=DEFAULT_RETRY)
+        try:
+            blob.upload_from_string(str(message).encode("utf8"), retry=DEFAULT_RETRY)
+        except Forbidden as e:
+            # If an object exists then the GCS Client will attempt to delete the existing object before reuploading.
+            # However, runner does not delete permission in an attempt to reduce duplicate receipts.
```

Comment:
"runner does not _**have**_ delete permission"?
"in an attempt to reduce duplicate receipts we didn't grant runner delete permission"?
You will know better anyway, just word missing.